### DISCUSSION TITLE
import.meta.glob incompatibility issue when migrating from Vite

### DISCUSSION THREAD

**Discussion by zhangfisher** [web-infra-dev/rsbuild#3106] [Q&A]
官方推荐使有`import.meta.webpackContext `替代`import.meta.glob`

但是在迁移`@generouted/react-router`插件时，碰到一个问题，其所依赖的`@generouted/core`使用了`import.meta.glob`
这下就麻烦了，本来只是想将`@generouted/react-router vite plugin`迁移，现在也得个修改依赖，显然不是一个好的方案

`@generouted/react-router` --> `@generouted/core` 

有什么好办法，能在不修改依赖j`@generouted/core`的情况下，解决这个兼容问题？

**Comment by chenjiahan** [comment]
There is a loader to transform Vite `import.meta` API to webpack styles, you can have a try: https://github.com/XiaoLin1995/import-meta-loader

**Comment by zhangfisher** [comment]
这是不是意味着得去重新编译`@generouted/core`

**Comment by zhangfisher** [comment]
由`rsbuild`同时支持`Vite import.meta API`是否更能彻底解决此问题？

否则，这意味着，我无法在重新修改编译`@generouted/core`的基础上，开发一个`rsbuild`版本的`@generouted/rsbuild`插件。

**Reply by chenjiahan** [reply]
`import.meta.glob` is a Vite-specific API, not a specification, and Rspack already supports webpack's `import.meta.webpackContext` API, so we are still considering whether to support `import.meta.glob`.

**Comment by zhangfisher** [comment]
如果能支持最好，否则就会存在一个传染性的问题。

由于一个依赖库使用了`Vite import.meta API`而放弃使用`rsbuild`

**Reply by chenjiahan** [reply]
In most cases, runtime libraries should avoid being tied to a specific build tool, and it is better to use standard APIs.

**Comment by zhangfisher** [comment]
`farm`倒是支持`vite import.meta.glob`语法，但是`glob`匹配算法不完全，像这样的语法不支持

```
var ROUTES = import.meta.glob(
  ["/src/pages/**/[\\w[-]*.{jsx,tsx,mdx}", "!/src/pages/**/(_!(layout)*(/*)?|_app|404)*"],
  { eager: true }
);
```
暂时放弃迁移计划
